enumerable.rb

```diff
@@ -0,0 +1,89 @@
+module Enumerable
+  def my_each
+    return to_enum unless block_given?
+
+    size.times { |n| yield (self[n]) }
+  end
+
+  def my_each_with_index
+    return to_enum unless block_given?
+
+    size.times { |n| yield self[n], n }
+  end
+
+  def my_select
+    return to_enum unless block_given?
+
+    result_array = []
+    my_each { |n| result_array << n if yield(n) }
+    result_array
+  end
+
+  def my_all
+    return true unless block_given?
+
+    my_each { |n| return false unless yield(n) }
+    true
+  end
+
+  def my_any
+    return true unless block_given?
+
+    my_each { |n| return true if yield(n) }
+    false
+  end
+
+  def my_none
+    return true unless block_given?
+
+    my_each { |i| return false if yield(i) == true }
+    true
+  end
+
+  def my_count(arg = nil)
+    counter = 0
+
+    if !arg.nil?
+      my_each { |i| counter += 1 if i == arg }
+    elsif block_given?
+      my_each { |i| counter += 1 if yield(i) }
+    else
+      counter
+    end
+    counter
+  end
+
+  def my_map(a_proc = nil)
+    return to_enum unless block_given?
+
+    map_array = []
+
+    if a_proc.nil?
+      my_each { |n| map_array << yield(n) }
+    else
+      my_each { |n| map_array << proc.call(n) }
+    end
+
+    map_array
+  end
+
+  # rubocop:disable Metrics/CyclomaticComplexity, Metrics/PerceivedComplexity
+  def my_inject(*args)
+    args_size = args.length
```

Comment:
### **my_inject**

- [ ] combines all elements of enum by applying a binary operation, specified by a block

- [ ] when a symbol is specified combines each element of the collection by applying the symbol as a named method

Reply:
the test passes all the test on the ruby documentation